src/main/java/com/spotify/docker/client/DockerClient.java

```diff
@@ -1003,6 +1001,39 @@ void restartContainer(String containerId, int secondsToWaitBeforeRestart)
   void killContainer(String containerId) throws DockerException, InterruptedException;
 
   /**
+   * Kill a docker container.
+   * @param containerId The id of the container to kill.
+   * @param signal Signal used to kill the process.
+   * @throws ContainerNotFoundException
+   *                              if container is not found (404)
+   * @throws DockerException      if a server error occurred (500)
+   * @throws InterruptedException If the thread is interrupted
+   */
+  void killContainer(final String containerId, final TerminationSignal signal)
+      throws DockerException, InterruptedException;
+
+  /**
+   * Supported parameters for {@link #killContainer(String, TerminationSignal)}).
+   */
+  enum TerminationSignal {
+    SIGINT("SIGINT"),
+    SIGQUIT("SIGQUIT"),
+    SIGTERM("SIGTERM"),
+    SIGKILL("SIGKILL"),
+    SIGSTOP("SIGSTOP");
+
```

Comment:
Here's what Dockers bash completion uses:
```bash
        local signals=(
                SIGCONT
                SIGHUP
                SIGINT
                SIGKILL
                SIGQUIT
                SIGSTOP
                SIGTERM
                SIGUSR1
                SIGUSR2
        )
```
This is quite an arbitrary subset of the available signals, but nobody ever complained so far.